ez/ez.go

```diff
@@ -41,7 +42,8 @@ func WithWatchingConfigFile() Option {
 	return func(d *dialsOptions) { d.watch = true }
 }
 
-type decoderFactory func(string) dials.Decoder
+// DecoderFactory should return the appropriate decoder based on the config file name
```

Comment:
nit: Let's extend his docstring slightly to say that the file path is passed as the string argument.